src/WhatFsBackend.js

```diff
@@ -0,0 +1,376 @@
+  async mkdir(filepath, opts) {
+    const paths = split(filepath)
+    const last = paths.length - 1
+    const dirname = paths[last]
+    const recursive = opts?.recursive || false
+    let firstCreated
+    let cursor = this._root
+
+    for (const i = 1; i <= last; ++i) {
+      const path = paths[i]
+      let existing
+      try {
+        existing = await cursor.getDirectoryHandle(path)
+      } catch(err) {
+      }
+
+      if (existing) {
+        if (i === last && !recursive) {
+          throw new Error(`Directory '${filepath}' already existed`)
+        }
+        cursor = existing
+      } else if (recursive) {
+        if (!firstCreated) {
+          firstCreated = paths.slice(0, i).join("/")
+        }
+        cursor = await cursor.getDirectoryHandle(path, { create: true })
+      } else if (i === last) {
+        cursor = await cursor.getDirectoryHandle(path, { create: true })
+      } else {
+        throw new Noent(filepath)
+      }
+    }
+    return recursive ? firstCreated : undefined
+  }
+  async rmdir(filepath, opts) {
+    if (opts?.recursive) {
+      throw new Error("Deprecated 'recursive' rmdir not impmlemented")
+    }
+    return this.unlink(filepath)
+  }
+  async rename(oldFilepath, newFilepath) {
+    // WANTED: https://github.com/whatwg/fs/pull/10
+    const content = await this.readFile(oldFilepath)
+    await this.writeFile(newFilePath, content)
+    // for safety sake putting this last, at cost of extra disk usage
+    await this.unlink(oldFilePath)
+  }
+  async stat(filepath, opts) {
+    if (opts?.bigint) {
+      throw new Error("Stat 'bigint' option not implemented")
+    }
+    const paths = split(filepath)
+    const filename = paths[paths.length - 1]
+    try {
+      const dir = await this._resolveDir(paths)
+      const handle = await dir.getFileHandle(filename)
+      const file = await handle.getFile()
+      const mtimeMs = file.lastModified
+      // NEEDED: https://github.com/whatwg/fs/issues/12
+      return {
+        size: file.size,
+        mtimeMs,
+        mtime: new Date(mtimeMs)
+      }
+    } catch(err) {
+      throw new Noent(filepath)
+    }
+  }
+  async lstat(filepath, opts) {
+    return this.stat(filepath)
+  }
+  async readlink(filepath, opts) {
+    // NEEDED: https://github.com/whatwg/fs/issues/54
+    throw new Error("Insufficient web standards for readlink");
+  }
+  async symlink(filepath, opts) {
+    // NEEDED: https://github.com/whatwg/fs/issues/54
+    throw new Error("Insufficient web standards for symlink");
+  }
+  async flush() {
+    // flush and cache would make sense if we kept a filepath->handle cache,
+    // which could definitely have other performance benefits
+  }
+  async close() {
+    // see `flush()` for some possibilities
+  }
+  async wipe() {
+    await this.rmdir("/", { recursive: true })
+  }
+  async watch() {
+    // NEEDED: https://github.com/WICG/file-system-access/issues/72
+    throw new Error("Insufficient web standards for watch")
```

Comment:
would love to see wicg/file-system-access#72 for watches so we could support this.